app/controllers/Janus.scala

```diff
@@ -98,7 +100,20 @@ class Janus(
       )
       loginUrl = Federation.loginUrl(credentials, host, stsClient)
     } yield {
-      SeeOther(loginUrl)
+      val redirectUrl = request.cookies.get("autoLogout") match {
```

Comment:
I'd love to see this logic be tested, which would also give us the opportunity to enforce some of the things you've discovered (liek the us-east-1 constrain). Could this logic move to the `logic` package, perhaps we can create a new object in there if there isn't a good existing candidate already. With this as a function we can write some tests that takes the "state" (currently from the cookie but see below) and confirm that we have returned the correct URL. If we do that, let's focus on the properties we're checking with a helpfully named test for each? I'll be happy to look at this with you if you'd like?

Additionally, I wonder whether this logic could be part of the work `Federation.loginUrl` achieves. That function returns the login URL, so it might make sense for it to take the "logout state" and return the correct URL based on that? This function performs side-effects (Amazon provide the signed console link from the credentials) so we'd still want this logic to be available elsewhere and tested, it's just a question of where it gets called. I think having the `loginUrl` on [line 101](https://github.com/guardian/janus-app/pull/448/files#diff-7ebc3def08c39769f2b0e152b9891084e6be013ffc4de7d8ea164fa571b845fbR101) be the answer from this controller's point of view would be a little clearer, but let me know what you think!

Reply:
In the PR that replaces this one, I've addressed this point by pulling out the redirect logic into a tested function. This also gives us the opportunity to add the auto-logout functionality to both the "console URL" and "console redirect" features without needing to duplicate the logic.